### PR TITLE
initial work on annotating nullable types for better Swift compatibility

### PR DESCRIPTION
**Pull request by damienpontifex**
Adding nullability annotations to MSClient relating to #669 . Only added to this class as a first stage and to receive feedback on these initial annotations and their usage.

I have run trials with this in my own Swift application and they seem be valid null/nonnull parameters. If these seem good and are accepted, can continue to markup other public classes in this trend.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@damienpontifex__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, AZPRBOT;

File: sdk/iOS/src/MSClient.h [hunk 1]
```diff
@@ -140,22 +140,22 @@
 #if TARGET_OS_IPHONE
 /// Logs in the current end user with the given provider by presenting the
 /// MSLoginController with the given controller.
--(void)loginWithProvider:(NSString *)provider
-              controller:(UIViewController *)controller
+-(void)loginWithProvider:(nonnull NSString *)provider
+              controller:(nonnull UIViewController *)controller
                 animated:(BOOL)animated
-              completion:(MSClientLoginBlock)completion;
+              completion:(nonnull MSClientLoginBlock)completion;
```
Comment (phvannor) [inline sdk/iOS/src/MSClient.h]: this can be null.

File: sdk/iOS/src/MSClient.h [hunk 2]
```diff
@@ -140,22 +140,22 @@
...
 
 /// Returns an MSLoginController that can be used to log in the current
 /// end user with the given provider.
--(MSLoginController *)loginViewControllerWithProvider:(NSString *)provider
-                                 completion:(MSClientLoginBlock)completion;
+-(nonnull MSLoginController *)loginViewControllerWithProvider:(nonnull NSString *)provider
+                                 completion:(nonnull MSClientLoginBlock)completion;
```
Comment (phvannor) [inline sdk/iOS/src/MSClient.h]: same for all the completion blocks.

File: sdk/iOS/src/MSClient.h [hunk 3]
```diff
@@ -168,14 +168,14 @@
 /// @{
 
 /// Returns an MSTable instance for a table with the given name.
--(MSTable *)tableWithName:(NSString *)tableName;
+-(nonnull MSTable *)tableWithName:(nonnull NSString *)tableName;
 
 /// Old method to return an MSTable instance for a table with the given name.
 /// This has been deprecated. Use tableWithName:
--(MSTable *)getTable:(NSString *)tableName __deprecated;
+-(nonnull MSTable *)getTable:(nonnull NSString *)tableName __deprecated;
```
Comment (phvannor) [inline sdk/iOS/src/MSClient.h]: might as well just delete this guy finally.

File: sdk/iOS/src/MSClient.h [hunk 4]
```diff
@@ -186,21 +186,21 @@
...
 /// response content will be treated as JSON.
--(void)invokeAPI:(NSString *)APIName
-            body:(id)body
-      HTTPMethod:(NSString *)method
-      parameters:(NSDictionary *)parameters
-         headers:(NSDictionary *)headers
-      completion:(MSAPIBlock)completion;
+-(void)invokeAPI:(nonnull NSString *)APIName
+            body:(nullable id)body
+      HTTPMethod:(nonnull NSString *)method
```
Comment (phvannor) [inline sdk/iOS/src/MSClient.h]: i believe this is allowed to be null, and if so, defaults to POST

File: sdk/iOS/src/MSClient.h [hunk 5]
```diff
@@ -140,22 +140,22 @@
 #if TARGET_OS_IPHONE
 /// Logs in the current end user with the given provider by presenting the
 /// MSLoginController with the given controller.
--(void)loginWithProvider:(NSString *)provider
-              controller:(UIViewController *)controller
+-(void)loginWithProvider:(nonnull NSString *)provider
+              controller:(nonnull UIViewController *)controller
```
Comment (phvannor) [inline sdk/iOS/src/MSClient.h]: did you check this one?  I'm not sure if this is required or not off the top of my head

Comment (damienpontifex) [inline sdk/iOS/src/MSClient.h]: If you don't provide the view controller, it cannot present the login controller. Hence why I chose nonnull on that one.

On the provider: the request generated in MSLogin appends login/<provider> to the gateway url. I would've though it wouldn't make sense to provide nil as the gateway would reject it as not knowing how to handle it? But I don't have the specific source code to confirm that for the backend.

File: sdk/iOS/src/MSClient.h (hunk 4 above)
Comment (damienpontifex) [inline sdk/iOS/src/MSClient.h]: True, it does in MSAPIRequest

**Pull request merged**